design/FY2022/enhancement_to_rainfall_handling.md

```diff
@@ -0,0 +1,55 @@
+1. *Site:Precipitation* object
+1. *RoofIrrigation* object
+
+Among them, rainfall flags in the weather file are used for the outside surface heat balance; *Site:Precipitation* is used for rainwater harvesting (in the *WaterUse:RainCollector* object) and the green roof; RoofIrrigation is used for the green roof; and precipitation amounts in the weather file are not used at all. 
+We propose the enhancement to integrate the use of weather file hourly precipitation amounts along with two existing input objects (*Site:Precipitation* and RoofIrrigation) to model rainwater harvesting and green roof runoff. If the *Site:Precipitation* values are presented, the values should be used. Otherwise, if there is no *Site:Precipitation* object, then values from the weather file should feed the *WaterUse:RainCollector* object and the green roof object.
+
+## Overview ##
+
+The *Site:Precipitation* values are used in two places in EnergyPlus, namely the rain collector and the roof irrigation. The rain collector, modeled by *WaterUse:RainCollector* object, is used for harvesting rainwater falling on building surfaces. The rainwater is then sent to a *WaterUse:Storage* object. Currently, the use of the rain collector object requires the inclusion of a *Site:Precipitation* object to describe the rates of rainfall. In the green roof module, the soil energy heat released or gained is calculated based on the phase changes of soil water, precipitation heat flux and heat flux due to vertical transport of water in the soil are ignored. The precipitation value is added to the surface soil moisture variable if a *Site:Precipitation* schedule exists. In particular, the green roof run off rate is calculated based on the irrigation rate on top of the precipitation rate.
+
+As local weather stations are providing hourly precipitation for the past decades and the values are being updated in TMY3 data, we propose to modify the logic of using rainfall/liquid precipitation to use the EPW file hourly precipitation data for rain harvesting and green roof when no *Site:Precipitation* schedule exists. In EPW files, the field of “Liquid Precipitation Depth” holds the amount of liquid precipitation (mm) observed at the indicated time for the period indicated in the liquid precipitation quantity field. Used in EnergyPlus simulation, if this value is not missing, it is used and overrides the “precipitation” flag as rainfall. Conversely, if the precipitation flag shows rain and this field is missing or zero, it is set to 1.5 (mm).
+
+## Approach ##
+
+When a *WaterUse:RainCollector* or a RoofIrrigation object is defined without a *Site:Precipitation* object presented, we will use the hourly precipitation value defined in the EPW file instead. When every entry of the liquid precipitation depth is missing in the 8760 hourly period, we will throw a warning and use the precipitation flag to estimate the precipitation quantity (as 1.5mm).
+
```

Comment:
I believe many EPW files are from sources where the rain amount is poorly estimated or unknown so I think there should be some protection for users in this case. I would suggest when no Site:precipitation object is present a warning message should be added. The message should provide the annual total precipitation that is shown in the EPW file. It would be even better if some guidance was provided in the warning message if this value in the EPW file is reasonable or not. Perhaps normal ranges based on the climate zone or climate type based on the stat file could provide guidance (although this is also often missing) or some other factor from the EPW file could indicate how wet that climate typically is.

Reply:
Good point. A warning message could be added to indicate the fact that the weather file will be used in RainCollector or RoofIrrigation calculation instead of Site:precipitation. Other warnings could also be included to flag out-of-normal-range rainfall data in the weather file.

Reply:
Hi @JasonGlazer, I added some descriptions in the approach section about the warning messages when EPW is used in RainCollection or Roof Irrigation calculation and when EPW precipitation data is out of range. The "out of range" is defined as the EPW hourly precipitation value being larger than the state recorded highest 24-hour precipitation, according to [this table](https://www.ncdc.noaa.gov/extremes/scec/records/all/maxp). I haven't found the hourly extreme yet, so I was considering using this 24-hour one as an initial upper bound. I'll see if I can find the hourly extreme record per state or climate zone and replace this upper bound later.

Reply:
Hi @JasonGlazer, just to confirm. Is it better to throw a warning when using epw in RainCollection or RoofIrrigation calculation, as well as provide a link of [this table](https://www.ncdc.noaa.gov/extremes/scec/records/all/maxp) in the documentation so that users can check the epw precipitation range before feeding it into the simulation? Or should I also include this table in the code and check the rainfall data range according to this table in EnergyPlus? I'm leaning towards the former as there might need some continuous effort in the future to maintain this table up-to-date. Any thoughts?